### PR TITLE
CB-10914 Create config provider for Cruise Control

### PR DESCRIPTION
**Pull request by patrik-marton**
Created a configuration provider for Cruise Control. It sets the self healing goals and their parameters which is specific in DataHub and cannot go as defaults into the CSD.

Update:
Commit 4e03a23 resolves a merge conflict in _CMRepositoryVersionUtil.java._
Commit d4413ed adds unit tests and modifies commit message as recommended

### CODE REVIEW

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/cruisecontrol/CruiseControlRoleConfigProvider.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+    @Override
+    public List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, HostgroupView hostGroupView, TemplatePreparationObject source) {
+        String cdpVersion = source.getBlueprintView().getProcessor().getStackVersion() == null ?
+                "" : source.getBlueprintView().getProcessor().getStackVersion();
+
+        if (isVersionNewerOrEqualThanLimited(cdpVersion, CLOUDERA_STACK_VERSION_7_2_11)) {
+            return List.of(config(SELF_HEALING_ENABLED, "true"),
+                    config(ANOMALY_NOTIFIER_CLASS, "com.linkedin.kafka.cruisecontrol.detector.notifier.SelfHealingNotifier"),
```
Comment (doktoric) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/cruisecontrol/CruiseControlRoleConfigProvider.java:75]: I think all of this should be a default variable which is setted on CM side.

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/cruisecontrol/CruiseControlRoleConfigProvider.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+import org.springframework.stereotype.Component;
+
+import com.cloudera.api.swagger.model.ApiClusterTemplateConfig;
+import com.sequenceiq.cloudbreak.cmtemplate.CmHostGroupRoleConfigProvider;
+import com.sequenceiq.cloudbreak.template.TemplatePreparationObject;
+import com.sequenceiq.cloudbreak.template.views.HostgroupView;
+
+@Component
+public class CruiseControlRoleConfigProvider implements CmHostGroupRoleConfigProvider {
```
Comment (doktoric) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/cruisecontrol/CruiseControlRoleConfigProvider.java:19]: unit test please

**Comment by doktoric** [comment]
commit message should be "CB-10914 <description with more detail>"

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/cruisecontrol/CruiseControlRoleConfigProvider.java (hunk 1 above)
Comment (benyoka) [reply template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/cruisecontrol/CruiseControlRoleConfigProvider.java:75]: We have different default configs for Private Cloud and DataHub. Private Cloud uses the upstream defaults which are the CM defaults as well and DataHub uses customized ones. This is why we need this config provider.

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/cruisecontrol/CruiseControlRoleConfigProvider.java (hunk 2 above)
Comment (patrik-marton) [reply template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/cruisecontrol/CruiseControlRoleConfigProvider.java:19]: Added unit test for the provider

**Comment by patrik-marton** [comment]
> commit message should be "CB-10914 description with more detail"

Done

**Review by doktoric — APPROVED** [review]

**Pull request merged**